sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java

```diff
@@ -129,6 +140,29 @@ public Plan consume(AnalyzedRelation relation, ConsumerContext context) {
         return bestOrder;
     }
 
+    /*
+     * Returns true if the join condition of the join pair actually includes both relations.
+     * @throws IllegalStateException when the join condition of the the join pair contains relations
+     *                               that are not already part of join tree.
+     */
+    private static boolean includesBothRelations(List<JoinPair> currentPermutationJoinPairs,
+                                                 JoinPair joinPair,
+                                                 QualifiedName rel1,
+                                                 QualifiedName rel2) throws IllegalStateException {
```

Comment:
@mfussenegger I'm not big fun of using Exceptions to change execution flow, so either I use a numeric return value or I don't extract the method at all, or better suggestion?